### PR TITLE
Refactor CollabCodeEdit-Server

### PR DESCRIPTION
**Pull request by PaulBrachmann**

### CODE REVIEW

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEcreateSessionWith.from..st [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
+actions
+CCEcreateSessionWith: anOrderedDictionary from: aCCERemoteClient
 	| sessionToken |
 
-	aClient identifier: (anOrderedDictionary at: #id).
+	aCCERemoteClient identifier: (anOrderedDictionary at: #id).
 	sessionToken := (self createSessionWith: anOrderedDictionary)
-		addCreator: aClient;
-		token.  
+		addCreator: aCCERemoteClient;
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEcreateSessionWith.from..st]: Why don't we take the Client as a second parameter to the method?

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEheartbeat.from..st [hunk 2]
```diff
@@ -1,4 +1,7 @@
-request handling
+actions
 CCEheartbeat: anOrderedDictionary from: aCCERemoteClient
-	" this delay solves the problem, that the server imediatly drops the connection when the message arrives after estblishing the connection"
-	self futureSend: #sendAckHeartbeat: at: 5000 args: ( Array newFrom: { aCCERemoteClient } )
\ No newline at end of file
+	"this delay solves the problem, that the server imediatly drops the connection when the message arrives after estblishing the connection"
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEheartbeat.from..st]: "estblishing"

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEjoinSession.from..st [hunk 3]
```diff
@@ -1,13 +1,13 @@
-request handling
-CCEjoinSession: anOrderedDictionary from: aClient
+actions
+CCEjoinSession: anOrderedDictionary from: aCCERemoteClient
 	| sessionToken |
 
 	sessionToken := anOrderedDictionary at: #token.
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEjoinSession.from..st]: Why do we need that tmp var here?

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/onError.from..st [hunk 4]
```diff
@@ -1,5 +1,5 @@
-event handling
-onError: anException from: aSocket
-	Transcript cr; show: aSocket.
+request handling
+onError: anException from: aCCERemoteClient
+	Transcript cr; show: aCCERemoteClient.
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/instance/onError.from..st:3]: We dont just throw the error?

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/onMessage.from..st [hunk 5]
```diff
@@ -1,12 +1,6 @@
...
-	
-	[self perform: ((CCEMessage prefix , message type) asSymbol) with: (message payload) with: aClient]
-			ifError: [ :ex :rcvr |
-					Transcript cr; show: 'Error handling message: ', ex.]
\ No newline at end of file
+request handling
+onMessage: someData from: aCCERemoteClient
+	self
+		performMessage: ([ (CCEMessage deserialize: someData) ]
+			ifError: [ :err :receiver | ^ self onError: err from: aCCERemoteClient ])
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/instance/onMessage.from..st]: Please use `error`

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/onMessage.from..st [hunk 6]
```diff
@@ -1,12 +1,6 @@
...
-		Transcript cr; show: 'Invalid message: ', someData.
-		^ self ].
-	Transcript show: 'server receive: ', message serialize asString; cr.
-	
-	[self perform: ((CCEMessage prefix , message type) asSymbol) with: (message payload) with: aClient]
-			ifError: [ :ex :rcvr |
-					Transcript cr; show: 'Error handling message: ', ex.]
\ No newline at end of file
+request handling
+onMessage: someData from: aCCERemoteClient
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/instance/onMessage.from..st]: `someData` does not sound squeaky

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/onRequest..st [hunk 7]
```diff
@@ -1,18 +1,13 @@
...
+request handling
+onRequest: aWebRequest
+	| remoteClient |
+
+	remoteClient := CCERemoteClient newFromSocket: aWebRequest asWebSocket.
+	remoteClient
+		timeout: self class defaultTimeOut;
+		onMessage: [ :data | self onMessage: data from: remoteClient ];
+		onClose: [ self onClose: remoteClient ];
+		onError: [ :ex | self onError: ex from: remoteClient ].
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/instance/onRequest..st]: Did you mean `exception`

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/performMessage.from..st [hunk 8]
```diff
@@ -0,0 +1,7 @@
+request handling
+performMessage: aCCEMessage from: aCCERemoteClient
+	[ self
+		perform: (CCEMessage prefix , aCCEMessage type) asSymbol
+		with: aCCEMessage payload
+		with: aCCERemoteClient ]
+		ifError: [ :err :receiver | ^ self onError: err from: aCCERemoteClient ]
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/instance/performMessage.from..st]: Well. There's an `err` again

File: packages/CollabCodeEdit-Server.package/CollabCodeEditServerServiceProvider.class/class/initialize.st [hunk 9]
```diff
@@ -0,0 +1,3 @@
+initialization
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CollabCodeEditServerServiceProvider.class/class/initialize.st]: The category for an initialize on class side should be `class initialization`

**Review by arminWel — CHANGES_REQUESTED** [review]
You should know me well enough: I always will find something quibble. But it wasn't easy.

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/onError.from..st (hunk 4 above)
Comment (PaulBrachmann) [reply packages/CollabCodeEdit-Server.package/CCEServer.class/instance/onError.from..st:3]: Definitely not. If one client messed something up, we don't want the whole server to come crashing down

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEcreateSessionWith.from..st (hunk 1 above)
Comment (PaulBrachmann) [reply packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEcreateSessionWith.from..st]: As I did not design the initial implementation, I can't really tell. I do however guess that creating a session w/o immediately assigning a creator may be useful in some future scenarios (e.g. recreation of a session after a server crash)

Comment (arminWel) [reply packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEcreateSessionWith.from..st]: Hmm. "may be useful in some future scenarios" sounds pretty like an antipattern. And we got CCO for that

**Review by arminWel — APPROVED** [review]

**Review by arminWel — CHANGES_REQUESTED** [review]
See discussion

File: packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEcreateSessionWith.from..st (hunk 1 above)
Comment (PaulBrachmann) [reply packages/CollabCodeEdit-Server.package/CCEServer.class/instance/CCEcreateSessionWith.from..st]: Agreed. Here we go...

**Review by arminWel — APPROVED** [review]
I'm never happy. But it's not as ugly as you are, so I will leave you alone.

**Pull request merged**